pkg/reconciler/awssnssource/subscription.go

```diff
@@ -228,6 +253,75 @@ func awsCredentials(cli coreclientv1.SecretInterface,
 	}, nil
 }
 
+// findSubscription returns the ARN of the subscription corresponding to the
+// given topic URL if it exists.
+func findSubscription(ctx context.Context, cli snsiface.SNSAPI, topicARN, endpointURL string) (string /*arn*/, error) {
+	in := &sns.ListSubscriptionsByTopicInput{
+		TopicArn: &topicARN,
+	}
```

Comment:
This function is the "entrypoint" of this PR.

It adds the possibility to perform a look up by SNS topic and endpoint URL in order to determine whether a subscription matching our criteria already exists.